common/src/main/java/com/microsoft/identity/common/internal/encryption/BaseEncryptionManager.java

```diff
@@ -0,0 +1,508 @@
+            }
+        }
+
+        // Loading key only once for performance.
+        if (mEncryptionKey == null) {
+            mEncryptionKey = loadKeyForEncryption();
+        }
+
+        final String blobVersion = mEncryptionKey.getKeyType().getBlobVersion();
+        Logger.verbose(TAG + methodName, "Encrypt version:" + blobVersion);
+        final byte[] blobVersionBytes = blobVersion.getBytes(AuthenticationConstants.ENCODING_UTF8);
+        final byte[] bytes = clearText.getBytes(AuthenticationConstants.ENCODING_UTF8);
+
+        // IV: Initialization vector that is needed to start CBC
+        final byte[] iv = new byte[DATA_KEY_LENGTH];
+        new SecureRandom().nextBytes(iv);
+        final IvParameterSpec ivSpec = new IvParameterSpec(iv);
+
+        // Set to encrypt mode
+        final Cipher cipher = Cipher.getInstance(CIPHER_ALGORITHM);
+        final Mac mac = Mac.getInstance(HMAC_ALGORITHM);
+        cipher.init(Cipher.ENCRYPT_MODE, mEncryptionKey.getKey(), ivSpec);
+
+        final byte[] encrypted = cipher.doFinal(bytes);
+
+        // Mac output to sign encryptedData+IV. Keyversion is not included
+        // in the digest. It defines what to use for Mac Key.
+        mac.init(mEncryptionKey.getHMACKey());
+        mac.update(blobVersionBytes);
+        mac.update(encrypted);
+        mac.update(iv);
+        final byte[] macDigest = mac.doFinal();
+
+        // Init array to store blobVersion, encrypted data, iv, macdigest
+        final byte[] blobVerAndEncryptedDataAndIVAndMacDigest = new byte[blobVersionBytes.length
+                + encrypted.length + iv.length + macDigest.length];
+        System.arraycopy(blobVersionBytes, 0, blobVerAndEncryptedDataAndIVAndMacDigest, 0,
+                blobVersionBytes.length);
+        System.arraycopy(encrypted, 0, blobVerAndEncryptedDataAndIVAndMacDigest,
+                blobVersionBytes.length, encrypted.length);
+        System.arraycopy(iv, 0, blobVerAndEncryptedDataAndIVAndMacDigest, blobVersionBytes.length
+                + encrypted.length, iv.length);
+        System.arraycopy(macDigest, 0, blobVerAndEncryptedDataAndIVAndMacDigest, blobVersionBytes.length
+                + encrypted.length + iv.length, macDigest.length);
+
+        final String encryptedText = new String(Base64.encode(blobVerAndEncryptedDataAndIVAndMacDigest,
+                Base64.NO_WRAP), AuthenticationConstants.ENCODING_UTF8);
+        Logger.verbose(TAG + methodName, "Finished encryption");
+
+        return getEncodeVersionLengthPrefix() + ENCODE_VERSION + encryptedText;
+    }
+
+    @Override
+    public synchronized String decrypt(final String encryptedBlob) throws GeneralSecurityException, IOException {
+        final String methodName = ":decrypt";
+        Logger.verbose(TAG + methodName, "Starting decryption");
+
+        if (StringExtensions.isNullOrBlank(encryptedBlob)) {
+            throw new IllegalArgumentException("Input is empty or null");
```

Comment:
Should we just return the `empty` or `null` string back here instead of throwing an exception ?